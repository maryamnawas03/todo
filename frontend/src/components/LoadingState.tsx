// src/components/LoadingState.tsx
import React from "react";

export const LoadingState: React.FC = () => {
  return (
    <div className="text-center py-12">
      <div className="flex justify-center mb-4">
        <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-purple-600"></div>
      </div>
      <p className="text-gray-500">Loading tasks...</p>
    </div>
  );
};
