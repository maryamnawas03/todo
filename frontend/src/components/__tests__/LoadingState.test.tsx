// src/components/__tests__/LoadingState.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LoadingState } from '../LoadingState';

describe('LoadingState', () => {
  describe('Rendering', () => {
    it('should render "Loading tasks..." message', () => {
      render(<LoadingState />);
      
      expect(screen.getByText('Loading tasks...')).toBeInTheDocument();
    });

    it('should be visible when rendered', () => {
      render(<LoadingState />);
      
      const loadingText = screen.getByText('Loading tasks...');
      expect(loadingText).toBeVisible();
    });

    it('should display loading indicator', () => {
      render(<LoadingState />);
      
      expect(screen.getByText(/loading/i)).toBeInTheDocument();
    });
  });

  describe('Content', () => {
    it('should have accessible loading indication', () => {
      render(<LoadingState />);
      
      const loadingMessage = screen.getByText('Loading tasks...');
      expect(loadingMessage).toBeInTheDocument();
    });

    it('should show appropriate message while fetching tasks', () => {
      render(<LoadingState />);
      
      expect(screen.getByText(/loading tasks/i)).toBeVisible();
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<LoadingState />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
