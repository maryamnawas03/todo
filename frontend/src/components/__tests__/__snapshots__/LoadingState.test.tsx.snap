// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingState Snapshot should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center py-12"
  >
    <div
      class="flex justify-center mb-4"
    >
      <div
        class="animate-spin rounded-full h-8 w-8 border-b-2 border-purple-600"
      />
    </div>
    <p
      class="text-gray-500"
    >
      Loading tasks...
    </p>
  </div>
</DocumentFragment>
`;
